fix(imports): prevent duplicate imports within the same file

murka opened the pull request:
Closes: #2939

murka committed:
fix(imports): prevent duplicate imports within the same file

anton-trunov commented on src/imports/resolveImports.ts:
```diff
@@ -45,6 +46,15 @@ export function resolveImports({
+            // Check for duplicates within the current file
+            if (resolvedPathsInCurrentFile.has(resolved.path)) {
+                throwCompilationError(
+                    `Duplicate import of '${resolved.path}' in file '${sourceFrom.path}'`,
```
Hey thanks for the fix! Here the error message would be a bit too long and contain the full path, let's make it shorter

verytactical replied:
Paths in error messages should be relative to `process.cwd()`

verytactical commented on src/test/compilation-failed/import-errors.spec.ts:
```diff
@@ -18,3 +18,16 @@ it("symlinks are not allowed", async () => {
+    });
+    expect(result.ok).toBe(false);
+    const message = result.error.map((err) => err.message).join("; ");
+    expect(message).toContain("Duplicate import of");
```
These tests would be a bit annoying to update. We prefer using `toMatchSnapshot` tests in such situations.

For cases when we need to check that compilation of a contract fails with a specific message, there's a helper `itShouldNotCompile` (see `src/test/compilation-failed/const-eval-failed.spec.ts`). If interface of `run` changes, we'll otherwise have to update this code too.

murka committed:
feat(imports): use relative paths in import errors and snapshot tests

- Show relative paths instead of absolute paths in duplicate import errors
- Update tests to use snapshots

murka committed:
Merge branch 'main' into fix/prevent-duplicate-imports
